examples/AWSDriverExample/src/main/java/software/amazon/ReadWriteSplittingSpringJdbcTemplateExample.java

```diff
@@ -0,0 +1,114 @@
+  private static final String DATABASE_NAME = "database-name";
+  private static final String USERNAME = "username";
+  private static final String PASSWORD = "password";
+  private static final int QUERY_LOOP_NUM = 32;
+
+  public static void main(String[] args) throws SQLException {
+//     scenario1();
+//     scenario2();
+    scenario3();
+  }
+
+  private static void scenario1() {
+    // given readonly not set and using simple postrges data source implementation
+    // expect no reader instance load balancing
+    DataSource dataSource = getSimplePostgresDataSource();
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+    for (int i=0; i < QUERY_LOOP_NUM; i++) {
+      jdbcTemplate.queryForObject("SELECT 1", Integer.class);
+    }
+  }
+
+  private static void scenario2() throws SQLException {
+    // given readonly set and using simple postrges data source implementation
+    // expect no reader instance load balancing
+    DataSource dataSource = getSimplePostgresDataSource();
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+    dataSource.getConnection().setReadOnly(true);
+    for (int i=0; i < QUERY_LOOP_NUM; i++) {
+      jdbcTemplate.queryForObject("SELECT 1", Integer.class);
+    }
+  }
+
+  private static void scenario3() throws SQLException {
+    // readonly set and using Hikari data source for connection pooling
+    // expect no reader instance load balancing
+    DataSource dataSource = getHikariCPDataSource();
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+    dataSource.getConnection().setReadOnly(true);
+    for (int i=0; i < QUERY_LOOP_NUM; i++) {
+      jdbcTemplate.queryForObject("SELECT 1", Integer.class);
+    }
+  }
+
+  private static DataSource getSimplePostgresDataSource() {
+    AwsWrapperDataSource ds = new AwsWrapperDataSource();
+    ds.setJdbcProtocol("jdbc:postgresql:");
+    ds.setDatabasePropertyName("databaseName");
+    ds.setServerPropertyName("serverName");
+    ds.setPortPropertyName("port");
+
+    ds.setTargetDataSourceClassName("org.postgresql.ds.PGSimpleDataSource");
+
+    Properties targetDataSourceProps = new Properties();
+    targetDataSourceProps.setProperty(
+        PropertyDefinition.PLUGINS.name,"readWriteSplitting,failover,efm");
+    targetDataSourceProps.setProperty(
+        ReadWriteSplittingPlugin.LOAD_BALANCE_READ_ONLY_TRAFFIC.name, "true");
+    targetDataSourceProps.setProperty("serverName", DATABASE_URL);
+    targetDataSourceProps.setProperty("databaseName", DATABASE_NAME);
+    targetDataSourceProps.setProperty("port", "5432");
+
+    ds.setUser(USERNAME);
+    ds.setPassword(PASSWORD);
+
+    ds.setTargetDataSourceProperties(targetDataSourceProps);
+
+    return ds;
+  }
+
+  private static DataSource getHikariCPDataSource() {
+    HikariDataSource ds = new HikariDataSource();
+
+    ds.setUsername(USERNAME);
+    ds.setPassword(PASSWORD);
+    ds.setMaximumPoolSize(2);
+    ds.setIdleTimeout(60000);
```

Comment:
Can you please make these values constants as well?

Reply:
Done